cram/cram_io.c

```diff
@@ -1893,15 +1893,29 @@ static int cram_populate_ref(cram_fd *fd, int id, ref_entry *r) {
     char *ref_path = getenv("REF_PATH");
     SAM_hdr_type *ty;
     SAM_hdr_tag *tag;
-    char path[PATH_MAX], path_tmp[PATH_MAX];
+    char path[PATH_MAX], path_tmp[PATH_MAX], cache[PATH_MAX];
     char *local_cache = getenv("REF_CACHE");
     mFILE *mf;
 
     if (fd->verbose)
 	fprintf(stderr, "cram_populate_ref on fd %p, id %d\n", fd, id);
 
-    if (!ref_path || *ref_path == 0)
+    if (!ref_path || *ref_path == 0) {
+	/*
+	 * If we have no ref path, we use the EBI server.
+	 * However to avoid spamming it we require a local ref cache too.
+	 */
 	ref_path = "http://www.ebi.ac.uk:80/ena/cram/md5/%s";
+	if (!local_cache || !*local_cache) {
```

Comment:
On Tue, Jan 27, 2015 at 06:57:27AM -0800, pd3 wrote:
> >  	ref_path = "http://www.ebi.ac.uk:80/ena/cram/md5/%s";
> > +	if (!local_cache || !*local_cache) {
> 
> On Tue, 2015-01-27 at 06:03 -0800, James Bonfield wrote:
> > In cram/cram_io.c:
> > 
> > >  	ref_path = "http://www.ebi.ac.uk:80/ena/cram/md5/%s";
> > > +	if (!local_cache || !*local_cache) {
> > 
> > It's a stylistic thing, but grepping around I see htslib and samtools
> > have many more uses of !*str than str[0] == '\0' so we may as well
> > keep it as is.
> 
> Beauty (and readability!) is in the eye of the beholder. I support
> James's and Heng's style, yet I am not insisting that others obey it.
> Can this be mutual please?

I don't really have a desparately consistent style for such things so
didn't mind changing it, but in principle I agree when it's in code
provided by one author like this (ie the cram subdirectory, or all the
vcf/bcf code in your case) that it should be largely up to the
developer in the absence of any overruling global style guide.

For shared code, it's probably best to have a bit more agreement on
keeping things to a common style.

We could of course insist that all code, whether from a single author
or mixed, adheres to the same conventions throughout (Picard does
this), but it is a big task and for now I'd prefer we tackle the more
urgent issues first before getting bogged down with purely stylistic
changes.

James

-- 
James Bonfield (jkb@sanger.ac.uk) | Hora aderat briligi. Nunc et Slythia Tova
                                  | Plurima gyrabant gymbolitare vabo;
  A Staden Package developer:     | Et Borogovorum mimzebant undique formae,
https://sf.net/projects/staden/   | Momiferique omnes exgrabure Rathi. 


-- 
 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.